src/apps/monero/signing/step_09_sign_input.py

```diff
@@ -126,28 +133,35 @@
         "Real source entry's mask does not equal spend key's",
     )
 
-    gc.collect()
-    state.mem_trace(4)
+    state.mem_trace(4, True)
+    mg_buffer = []
 
     from apps.monero.xmr import mlsag
 
     if state.rct_type == RctType.Simple:
         ring_pubkeys = [x.key for x in src_entr.outputs]
-        mg = mlsag.generate_mlsag_simple(
+        src_entr = None
+
```

Comment:
I was thinking about this before already and I'd suggest to change this to:
```
ring_pubkeys = [x.key for x in src_entr.outputs]
src_entr = None
```

before the `if` and then modify `mlsag.generate_mlsag_full` to accept the `ring_pubkeys` in the same way as mlsag_simple.